readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -61,6 +61,18 @@ def __init__(self, *args, **kwargs):
                     self.config_file,
                 )
         except ProjectConfigurationError:
+            # NOTE: this exception handling here is weird to me.
+            # We are raising this exception from inside `project.confi_file` when:
+            #  - the repository has multiple config files (none of them with `doc` in its filename)
+            #  - there is no config file at all
+            #
+            # IMO, if there are multiple config files,
+            # the build should fail immediately communicating this to the user.
+            # This can be achived by unhandle the exception here
+            # and leaving `on_failure` Celery handle to deal with it.
+            #
+            # In case there is no config file, we should continue the build
+            # because Read the Docs will automatically create one for it.
```

Comment:
This seems reasonable. Is there a good way to implement the suggested logic? I guess raise a different exception for multiple configs that we `raise` (or just don't catch)?

Reply:
Yes. I documented this here in this way because I want to change the internal behavior and also communicate these errors better to the user. I'm opening a new issue with all these findings to continue working after this, since it will require some refactor and more design decisions.